src/Storefront/Checkout/Cart/RestoreUrlService/RestoreUrlService.php

```diff
@@ -22,16 +23,21 @@ class RestoreUrlService
     public function __construct(
         private readonly EntityRepository $mappingRepository,
         private readonly UrlGeneratorInterface $urlGenerator,
+        private readonly ConfigProvider $configProvider,
         private readonly RequestStack $requestStack,
     ) {
     }
 
-    public function getCurrentRestoreUrl(SalesChannelContext $context): string
+    public function getCurrentRestoreUrl(SalesChannelContext $context): ?string
     {
-        $current = $this->fetchFromDb($context->getToken(), $context->getContext());
-        return $this->generate(
-            $current ? $current->getId() : $this->createNew($context->getToken(), $context->getContext()),
-        );
+        if ($this->configProvider->isEnabledStoreDataAbundantCarts()) {
+            $current = $this->fetchFromDb($context->getToken(), $context->getContext());
+            return $this->generate(
+                $current ? $current->getId() : $this->createNew($context->getToken(), $context->getContext()),
+            );
+        }
+
+        return null;
```

Comment:
have you checked that there are no issues with ev1 requests etc (there shouldn't be, but might be a good idea to check)

Reply:
Here is an example of what it looks like when the user does not enable the "Store data for abandoned carts" option and this function returns null.
![image](https://github.com/user-attachments/assets/6f337b73-c289-4001-a6a3-e30b5cbebf38)

Reply:
I think thats ok :) if everything else works we are good